lib/dolly/document.rb

```diff
@@ -42,10 +42,16 @@ class Document
 
     def initialize(attributes = {})
       properties.each(&build_property(attributes))
+      send(slug_callback) if slugable?
```

Comment:
What is the purpose of this?

Reply:
IMO checking this for every Dolly document even when it is not slugable sounds weird. What about defining the initialize in the slugable module.

```ruby
module Dolly
  module Slugable
    def self.included(base)
      base.extend(ClassMethods)
    end
 
    def initialize(attributes = {})
      super(attributes)
      send(slug_callback) if slugable?
    end

   ...
  end
end
```

You can remove the sluggable? method or just move it to the module and return true if you want to know if the document is slugeable.

Reply:
Even that, I believe this should not be part of dolly repo. I know that making another gem requires more time but for me it is a better solution. Imagine this:

```ruby
gem 'dolly'
gem 'dolly-slugeable'
```

For me sounds a better approach and we keep dolly simple and small.

Reply:
@javierg @ErickFabian What do you think?

Reply:
Not for now, we can separate it later, but now we need it included, is still not a lot of code, and is very important on what we do now